prow/cmd/hook/main.go

```diff
@@ -152,14 +155,25 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting plugins.")
 	}
 
+	metrics, err := hook.NewMetrics()
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to initialize metrics.")
+	}
+
+	if err := push.FromGatherer("hook", push.HostnameGroupingKey(), configAgent.Config().PushGateway.Endpoint, prometheus.DefaultGatherer); err != nil {
```

Comment:
We should run this only when the endpoint is defined?

Reply:
This thing will go and push to Prometheus? Do we have anything to push at this point?

Reply:
At the moment, we'll push the webhooks counter.

Reply:
I mean, we haven't received a single webhook at this point so the metric is empty, right? FromGatherer seems to collect all the existing metrics and pushes to the endpoint via a single request. It seems that you need to run this in a separate goroutine that loops on an interval.

Reply:
Ah, I see. I was under the assumption that it would create a thread to periodically push.